torch/utils/data/datasets/readfilesfromtardataset.py

```diff
@@ -0,0 +1,58 @@
+from torch.utils.data.dataset import IterableDataset
+from torch.utils.data.datasets.common import validate_pathname_binary_tuple
+from typing import Iterable, Iterator
+
+import os
+import tarfile
+import warnings
+
+class ReadFilesFromTarIterableDataset(IterableDataset):
+    r""" :class:`ReadFilesFromTarIterableDataset`.
+
+    IterableDataset to extract tar binary streams from input iterables
+    yield pathname and extracted binary stream in a tuple.
+    args:
+        dataset: Iterable dataset that provides pathname and tar binary stream in tuples
+        length: a nominal length of the dataset
+    """
+    def __init__(
+            self,
+            dataset : Iterable,
+            length : int = -1):
+        super().__init__()
+        self.dataset : Iterable = dataset
+        self.length : int = length
+
+    def __iter__(self) -> Iterator[tuple]:
+        if not isinstance(self.dataset, Iterable):
+            warnings.warn("dataset must be Iterable type but got {}".format(type(self.dataset)))
+            raise TypeError
+        for data in self.dataset:
+            ret = validate_pathname_binary_tuple(data)
+            if ret:
+                warnings.warn("got invalid pathname and binary record ({}), abort!".format(ret))
+                raise TypeError
+            try:
+                tar = tarfile.open(fileobj=data[1], mode="r:*")
+                for tarinfo in tar:
+                    if not tarinfo.isfile():
+                        continue
```

Comment:
Curious if it is not the file, what it is? Maybe reason to error?

Reply:
The original purpose is to bypass dir